### PR TITLE
Completely bans sleep from Initialize and LateInitialize

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
So anyway I started cooking.

This completely bans `sleep()` from `Initialize()` and `LateInitialize()`. If you **REALLY** need it, "thread" it off with `addtimer()` to happen post-init and when the MC is firing.

A few highlights of this PR that snuck in:
- Optimised chemsmoke
- Removed `/obj/machinery/kitchen` - been broke for 10 years
- Made depot consoles not try and use `tgui_input_list` on init when theres no player (dont ask)
- Fixed some random bogus calls
- Removed some random delays in places that really didnt add much to the game
- If you merge this without testmerging you have more "FUCK IT SEND IT" rights than you should

To give you an idea how ludicrous these chains can be:
- Initializing punpun equips his jumpsuit
- Equiping anything *could* call `attackby()` on the item
- This could make its way up to the parent call
- Which could then go down anywhere through object
- And prompt for a pen input on Initialize().
- I need more vodka

## Why It's Good For The Game
`Initialize()` shouldnt sleep. This is my way of outright banning it and making sure it can never happen.

## Testing
I tried chemsmoke and paper bundles and some other stuff but this really needs a TM.

## Changelog
:cl: AffectedArc07
add: Optimised some init stuff
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/depot_consoles.dm [hunk 1]
```diff
@@ -508,9 +500,12 @@
 	if(!portal_enabled && myportal)
 		to_chat(user, "<span class='notice'>Outgoing Teleport Portal: deactivating... please wait...</span>")
 		return
-	toggle_portal()
+
+	var/turf/tele_target = choosetarget()
+	if(!tele_target)
+		return
+	toggle_portal(tele_target)
```
Comment (hal9000PR) [inline code/game/machinery/computer/depot_consoles.dm:507]: `toggle_portal` doesnt take any args. I think you meant to update it on line 511 to accept this as an arg and pass it through to `update_portal` on line 513

File: code/game/objects/effects/effect_system/effects_chem_smoke.dm [hunk 2]
```diff
@@ -94,22 +98,21 @@
...
+		new /obj/effect/particle_effect/chem_smoke/small(location, eff_colour)
+	else
+		new /obj/effect/particle_effect/chem_smoke(location, eff_colour)
 
-		sleep(1)
-	qdel(src)
+	INVOKE_ASYNC(src, PROC_REF(SmokeEm), eff_range)
+	if(process_count > max_process_count)
+		STOP_PROCESSING(SSprocessing, src)
+		qdel(src)
```
Comment (hal9000PR) [inline code/game/objects/effects/effect_system/effects_chem_smoke.dm:115]: does having the rate that this process's have any in game effect?

File: code/modules/library/book.dm [hunk 3]
```diff
@@ -360,11 +360,14 @@
 	var/amount = 1
 
 /obj/item/book/random/Initialize()
-	..()
-	var/list/books = GLOB.library_catalog.get_random_book(amount, FALSE)
+	. = ..()
+	addtimer(CALLBACK(src, PROC_REF(spawn_books)), 0)
```
Comment (hal9000PR) [inline code/modules/library/book.dm:364]: why a timer with delay 0 instead of an INVOKE_ASYNC?

**Review by hal9000PR — CHANGES_REQUESTED** [review]
just a few things.

theres a few 0 delay timers which I am not sure about, whats the reason for their use?

**Comment by AffectedArc07** [comment]
> just a few things.
> 
> theres a few 0 delay timers which I am not sure about, whats the reason for their use?

image

I'll address everything else today

File: code/modules/library/book.dm (hunk 3 above)
Comment (AffectedArc07) [reply code/modules/library/book.dm:364]: Explained on discord

File: code/game/objects/effects/effect_system/effects_chem_smoke.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/game/objects/effects/effect_system/effects_chem_smoke.dm:115]: It updates the visual a little slower but still does the same amount of injecting - I think

File: code/game/machinery/computer/depot_consoles.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/game/machinery/computer/depot_consoles.dm:507]: Updated in next commit

**Review by hal9000PR — APPROVED** [review]
what could go wrong

**Pull request merged**